nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -219,15 +221,41 @@ public void createNamespace(Namespace namespace, Map<String, String> metadata) {
     checkNamespaceIsValid(namespace);
     getRef().checkMutable();
     ContentKey key = ContentKey.of(namespace.levels());
-    org.projectnessie.model.Namespace content =
-        org.projectnessie.model.Namespace.of(key.getElements(), metadata);
     try {
       Content existing = api.getContent().reference(getReference()).key(key).get().get(key);
       if (existing != null) {
         throw namespaceAlreadyExists(key, existing, null);
       }
+
+      // check if the parent namespace exists
+      List<ContentKey> keys = Lists.newArrayList();
+      Map<ContentKey, Content> contentInfo = Maps.newHashMap();
+      for (int i = 0; i < namespace.levels().length; i++) {
+        Namespace parent = Namespace.of(Arrays.copyOf(namespace.levels(), i + 1));
+        ContentKey parentKey = ContentKey.of(parent.levels());
+        keys.add(parentKey);
+        Content parentContent =
+            api.getContent().reference(getReference()).key(parentKey).get().get(parentKey);
+        contentInfo.put(parentKey, parentContent);
+      }
+
+      List<Operation.Put> putOperations = Lists.newArrayList();
+      for (Map.Entry<ContentKey, Content> contentKeyContentEntry : contentInfo.entrySet()) {
+        if (contentKeyContentEntry.getValue() == null) {
+          org.projectnessie.model.Namespace content;
+          if (contentKeyContentEntry.getKey().equals(key)) {
+            content = org.projectnessie.model.Namespace.of(key.getElements(), metadata);
+          } else {
+            content =
+                org.projectnessie.model.Namespace.of(
+                    contentKeyContentEntry.getKey().getElements(), ImmutableMap.of());
+          }
+          putOperations.add(Operation.Put.of(contentKeyContentEntry.getKey(), content));
+        }
+      }
+
       try {
-        commitRetry("create namespace " + key, Operation.Put.of(key, content));
+        commitRetry("create namespace " + keys, putOperations.toArray(new Operation.Put[0]));
```

Comment:
The commit message would say that it created all parent namespaces, including those that already existed.